runtime/src/bank.rs

```diff
@@ -2615,6 +2622,49 @@ impl Bank {
         }
     }
 
+    pub fn calculate_capitalization(&self) -> u64 {
+        self.get_program_accounts(None)
+            .into_iter()
+            .filter_map(|(_pubkey, account)| {
+                if account.lamports == u64::max_value() {
+                    return None;
+                }
+
+                let is_specially_retained = solana_sdk::native_loader::check_id(&account.owner)
+                    || solana_sdk::sysvar::check_id(&account.owner);
+
+                if is_specially_retained {
+                    // specially retained accounts are ensured to exist by
+                    // alwaysing having a balance of 1 lamports, which is
+                    // outside the capitalization calculation.
+                    Some(account.lamports - 1)
+                } else {
+                    Some(account.lamports)
+                }
+            })
+            .sum()
+    }
+
+    pub fn reset_with_recalculated_capitalization(&self) -> u64 {
+        let old = self.capitalization();
+        self.capitalization
+            .store(self.calculate_capitalization(), Ordering::Relaxed);
+        old
+    }
+
+    pub fn assert_capitalization(&self) {
```

Comment:
I feel like this function belongs in `ledger-tool/`, it's not used elsewhere right?